game/views.py

```
@@ -339,16 +339,37 @@ def ai_move(request):
     depth_map = {'easy': 1, 'medium': 2, 'hard': 3}
     depth = depth_map.get(difficulty, 2)
 
-    best = game.get_ai_move(depth=depth)  # called once only
-    
+
+    # Prevent unhandled AI engine exceptions from crashing the endpoint with a 500 error
+    try:
+        best = game.get_ai_move(depth=depth)
+    except Exception as exc:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)